package.json

```diff
@@ -10,9 +10,10 @@
     "@types/jest": "^26.0.14",
     "jest": "^26.4.2",
     "marked": "^0.4.0",
-    "prettier": "^2.1.2",
+    "prettier": "^2.7.1",
```

Comment:
Prettier wasn't formatting the `_paths.ts` file correctly until I updated. I suspect it was the `infer Tail extends Key[]` syntax.

Reply:
i just merged a PR that enforces prettier checks in the pipeline.
if we want to update prettier, I think it would be better to create a separate PR that just updates prettier and runs prettier formatting on the whole codebase. It would avoid cluttering this PR with unrelated formatting changes. Could you do that?

Reply:
I could do that. But I'm also wondering if I should be using this syntax at all. It would mean consumers of `rememda` would need to be on TS 4.7 or greater which would be a breaking change.

Reply:
I might spend some time testing this -- it might be possible to document "this helper requires TS 4.7+" rather than requiring the whole lib to do that.

Reply:
yeah, what we could do is could ship differential types. We used to do that for [react-query v3](https://github.com/TanStack/query/blob/5e37b145a8836b00dd20860e8eb1f94c608fec81/package.json#L83-L85) when we introduced something that used variadic tuples but we also wanted to support versions < 4.1

Question is if we could come up with a type for this helper that would work well enough with TS < 4.7, or maybe, it could just accept `Array<string>` and we document that it only works "well" with TS 4.7 and above.

Reply:
I'm pretty sure `infer extends` is just a shorthand so we could ship the same type essentially.

```ts
Foo extends Array<infer U extends string>
  ? 'foo'
  : never

// Is equivalent to

Foo extends Array<infer U>
  ? U extends string
    ? 'foo'
    : never
  : never
```

The types get less readable but with the same functionality.

Reply:
that would also be fine 👍